src/sidebar/helpers/mentions.ts

```diff
@@ -0,0 +1,109 @@
+import type { Mention } from '../../types/api';
+
+/**
+ * Wrap all occurrences of @mentions in provided text into the corresponding
+ * special tag, as long as they are surrounded by "empty" space (space, tab, new
+ * line, or beginning/end of the whole text).
+ *
+ * For example: `@someuser` with the `hypothes.is` authority would become
+ *  `<a data-hyp-mention data-userid="acct:someuser@hypothes.is">@someuser</a>`
+ */
+export function wrapMentions(text: string, authority: string): string {
+  return text
+    .replace(/(?:^|\s)@(\w+)(?=\s|$)/g, (match, username) => {
+      const tag = document.createElement('a');
+
+      tag.setAttribute('data-hyp-mention', '');
+      tag.setAttribute('data-userid', `acct:${username}@${authority}`);
+      tag.textContent = `@${username}`;
+
+      return ` ${tag.outerHTML}`;
+    })
+    .trim();
+}
+
+/**
+ * Replace all mentions wrapped in the special `<a data-hyp-mention />` tag with
+ * their plain-text representation.
+ */
+export function unwrapMentions(text: string) {
+  const tmp = document.createElement('div');
+  tmp.innerHTML = text;
+  for (const node of tmp.querySelectorAll('a[data-hyp-mention]')) {
+    node.replaceWith(node.textContent ?? '');
+  }
+  return tmp.innerHTML;
+}
+
+export type InvalidUsername = string;
+
+function elementForMention(
+  mentionLink: HTMLElement,
+  mention?: Mention,
+): [HTMLElement, Mention | InvalidUsername] {
```

Comment:
This function sets some styles to the generated elements, via tailwind classes or style property.

They should be considered temporary, as we probably need to eventually align on the actual look and feel, specially for invalid mentions, which were not covered in the design.

Reply:
We could consider using regular CSS to style mentions, using an `a[data-hyp-mention]` selector for example, rather than Tailwind properties.

Another option would be to pass the classes used for different mention states as part of an options object.